### DISCUSSION TITLE
HTTPS connection unreliable when Binding Host set

### DISCUSSION THREAD

**Discussion by alainkovacs** [coddingtonbear/obsidian-web#169] [Bug Report]
I've just discovered this extension about a week ago, and my experience with it is very spotty. It's a really great idea and addon to Obsidian, but I've had more troubles than success with the extension unfortunately, so I'm hoping to find a solution to my problems.

TLDR; I can't get it to work with HTTPS, only works sometimes with HTTP, can't figure out why.

I can't be sure of course, but it looks like a bug to me. The app icon reports an error (and it can't be used - pic2) but when I go to settings I see this
![image](https://github.com/coddingtonbear/obsidian-web/assets/33669954/4add5875-4c22-40df-824e-9ee5a8666c83)
![image](https://github.com/coddingtonbear/obsidian-web/assets/33669954/09bdaa20-268f-4eb4-b46c-f51d05c130f2)
![image](https://github.com/coddingtonbear/obsidian-web/assets/33669954/8555388e-46ac-48d9-9ae1-c9339c439759)
![image](https://github.com/coddingtonbear/obsidian-web/assets/33669954/cd19bc06-0c45-4d1e-8a8e-bbdef2dd1a2a)

What I did initially is install the Local REST API, install the certificate, install chrome extension. It did not work from the start. It started working by enabling HTTP and instead of 127.0.0.1 using localhost. I don't understand why as those should be the same... Then it stopped working after a day or two (without me modifying/doing anything). I have tried installing the certificate again, disabling the HTTP, without success unfortunately.

I have not modified any advanced settings, ports or anything like that. What am I doing wrong?

**Comment by coddingtonbear** [comment]
You know, actually, I've been having a little bit of trouble with this, too, over the last few days on *one* of the two computers I access my vault from, and I'd thought that maybe it was just some quirk of my environment.

I tried digging into it in a little depth last night and couldn't quite figure out what was going on -- all I've found is the `TypeError: Failed to fetch` in my logs with little additional information, and strangely, as far as I can tell the request that shows you (and me) that green checkbox is identical to the ones that are failing.  If you're seeing that, too, I'm a little bit suspicious that maybe there's some kind of permissions problem -- there are so many overlapping permissions checks right now and maybe one of them is clobbering our permissions to reach localhost?

In any case: I think the short-term workaround would be to just use HTTP, unfortunately.  I'll try to see if I can figure out what's going wrong this weekend.

**Reply by crokobit** [reply]
Is there a way to configure the current extension to utilize HTTP? I've been unable to find an option for this. @coddingtonbear 🙏

**Reply by coddingtonbear** [reply]
Oh, hrm... I was about to say that you could do the toggle in `obsidian-local-rest-api`, but...yeah, now that I'm thinking about it, that toggles the opposite thing.

Ok, new plan:  you can make the HTTPS server inaccessible to obsidian-web by going to the settings for obsidian-local-rest-api and changing the port number in use for HTTPS to something unexpected (I've just tested this by changing it to `27125` from its usual value of `27124`).

**Reply by alainkovacs** [reply]
Haha! Just found exactly the same workaround(even used the same port)  a few hours ago, didn't have time to post about it, you beat me to it. From the tests I've done, it seems stable for now. 

Maybe one more thing to mention that I've not mentioned before, is that I actually have 2 instances of chrome (one for work, one private), but I can't see how this would cause this problem (I've had the same issues as described above with only one instance opened too). 

Regarding permissions, I have accepted all permissions that I was asked to accept.

**Comment by coddingtonbear** [comment]
OK!  I think I figured out what's going wrong and have updated the Obsidian Local REST API Obsidian plugin to fix it.  The problem is described here: https://github.com/coddingtonbear/obsidian-local-rest-api/releases/tag/2.1.1 if you're super curious.

To get and apply this fix:

1. Go into Obsidian and make sure your community plugins are up-to-date.  In this case, you're looking for version 2.1.1 -- that's the one with this fix.
2. Re-generate certificates by going into the Obsidian Local REST API plugin settings and pressing the "Re-generate Certificates" button.
3. Download the certificate again from that same page, and install it into your browser as a trusted certificate authority (CA) via whatever mechanism your OS and browser combination uses.

I should say, this fix seems to have fixed _my_ trouble around this, but if I understand the problem correctly, it _should_ only have been a problem for folks who had set the "Binding Host" field in Obsidian Local REST API's advanced settings.  In my case, I had set it to `0.0.0.0` so I could reach my obsidian instance from other hosts, and I'm curious to hear if those of you who did have this problem did the same thing.

Good luck!

**Reply by alainkovacs** [reply]
Thanks for the very fast reaction and fix! It does seem to fix the problem indeed. Though it was a bit less straightforward. I will leave my experience/solution below, should anybody encounter the same problems as me.

I'm on Win 10, running obsidian 1.5.3. After updating the REST API to 2.1.1, when I tried to regenerate the certificates as you mention, the plugin seemed to crash in an odd way and since it was crashing when I pressed the re-generate certs button, I tried resetting all crypto, which fully crashed the plugin. The "options" button was not showing up, even after restarting/enabling-disabling the plugin. I ended up re-installing the Local REST API plugin and started from scratch. 
After downloading the new certificate, I would like to stress the importance of installing it as a trusted CA. Before, I just pressed the "next/next", but now I followed your advice, and instead of letting windows choose where to install the certificate, I chose to select where the certificate is placed
![image](https://github.com/coddingtonbear/obsidian-web/assets/33669954/4bf8f575-a75e-4c03-8493-08171d369adf)
I am wondering though, what kind of impact this had, and why. I only understand to a limited degree how these certificates work, I would think that it would work no matter in which "store" you place it.. Anyways, the good part is that it works!
